src/test/objectstore/allocator_replay_test.cc

```diff
@@ -389,5 +395,104 @@ int main(int argc, char **argv)
                   << ", unit:" << alloc_unit << std::endl;
         return 0;
       });
+  } else if (strcmp(argv[2], "replay_alloc") == 0) {
+    if (argc < 4) {
+      std::cerr << "Error: insufficient arguments for \"replay_alloc\" option."
+                << std::endl;
+      usage_replay_alloc(argv[0]);
+      return 1;
+    }
+    return replay_free_dump_and_apply(argv[1],
+      [&](Allocator *a, const string &aname) {
+        ceph_assert(a);
+        std::cout << "Fragmentation:" << a->get_fragmentation()
+                  << std::endl;
+        std::cout << "Fragmentation score:" << a->get_fragmentation_score()
+                  << std::endl;
+        std::cout << "Free:" << std::hex << a->get_free() << std::dec
+                  << std::endl;
+        {
+          /* replay a set of allocation requests */
+          char s[4096];
+
+          FILE *f_alloc_list = fopen(argv[3], "r");
+          if (!f_alloc_list) {
+            std::cerr << "error: unable to open " << argv[3] << std::endl;
+            return -1;
+          }
+
+          /* Replay user specified number of times to simulate extended activity
+           * Defaults to 100 replays.
+           */
+          auto replay_count = 100;
+          if (argc == 5) {
+            replay_count = atoi(argv[4]);
+          }
+
+          for (auto i = 0; i < replay_count; ++i) {
+            while (fgets(s, sizeof(s), f_alloc_list) != nullptr) {
+              /* parse allocation request */
+              uint64_t want = 0, unit = 0, max = 0, hint = 0;
+
+              if (std::sscanf(s, "%ji %ji %ji %ji", &want, &unit, &max, &hint) < 2)
+              {
+                cerr << "Error: malformed allocation request:" << std::endl;
+                cerr << s << std::endl;
+                /* do not attempt to allocate a malformed request */
+                continue;
+              }
+
+              /* timestamp in milliseconds (before) */
+              auto ts1 = std::chrono::system_clock::now();
```

Comment:
you might want to use ceph::mono_clock() from ceph_time.h instead - it would result in a fewer lines ;)
auto t0 = ceph::mono_clock::now();
...
cout << ceph::mono_clock::now() - t0 << .....

Reply:
Thanks for the pointer, done!